src/Tribe/Admin/Settings.php

```diff
@@ -221,6 +222,9 @@ public function add_admin_pages() {
 			]
 		);
 
+		// Redirects users from any TEC setting to the Guided Setup page on first visit.
+		do_action( 'tec_redirect_guided_time_setup' );
```

Comment:
```suggestion
		do_action( 'tec_admin_redirect_guided_time_setup' );
```
A suggestion for adding 'admin_' to the prefix, which feels more in-line with our naming standards. @Camwyn?

Reply:
Also needs full docblock.

Actions and filters are documented similar to a function/method would.

/**
 *
 */

Reply:
I'll also explain my previous comment about why this place is not ideal for redirects and why i would prefer the `current_screen` hook.

Your custom action is firing along with the add_admin_pages method which is being fired along in the `admin_menu` action.

This is all fine currently.

BUT the method you are adding has one more hidden dependency: html output to not have started (you can't send headers and as a result redirect a user when html output has started)

This is being served by the current `admin_menu` hook. In my mind always lingers a though what if somebody changes the hook for the add_admin_pages method and as a result for my custom action as well ? My redirect could break then.

That's why for admin side redirects i choose for my callback to be directly bound to the `current_screen` hook. This is the last hook before html output starts while WP core has already gathered all the info you might need to evaluate your conditionals. (current screen has been fully set).

Reply:
For FE redirects we should be bound to `template_redirect` hook ;) Just a tip, not relevant here.